cmd/geth/main.go

```diff
@@ -400,6 +401,9 @@ func run(ctx *cli.Context) {
 
 	ethereum, err := eth.New(cfg)
 	if err != nil {
+		if datadirInUseErrRE.MatchString(err.Error()) {
+			err = fmt.Errorf("%v (another instance of geth is already running with the same data directory '%v')", err, ctx.GlobalString(utils.DataDirFlag.Name))
+		}
```

Comment:
This doesn't work. The returned error from `eth.New` is the original error message with the database which could not be opened prefix. You can change the regex but a better solution would be to catch the error in `ethdb.NewLDBDatabase` and convert it to an error with a more user friendly message. That would save a string comparison. You can test if the returned error in `leveldb.OpenFile` is `syscall.Errno` (11).

The last argument in `Errorf` is a string, use the `%s` formatter.